tests/waku_relay/test_protocol.nim

```diff
@@ -0,0 +1,1288 @@
+
+      discard node.subscribe(pubsubTopic, simpleFutureHandler)
+      discard otherNode.subscribe(pubsubTopic, otherSimpleFutureHandler)
+      check:
+        node.subscribedTopics == pubsubTopicSeq
+        otherNode.subscribedTopics == pubsubTopicSeq
+      await sleepAsync(500.millis)
+
+      # When sending multiple messages from node
+      let
+        msg1 = fakeWakuMessage("msg1", pubsubTopic)
+        msg2 = fakeWakuMessage("msg2", pubsubTopic)
+        msg3 = fakeWakuMessage("msg3", pubsubTopic)
+        msg4 = fakeWakuMessage("msg4", pubsubTopic)
+      
+      discard await node.publish(pubsubTopic, msg1)
+      # FIXME: Test SEGFAULTS when publishing after first publish in pubsub.nim:handleData:317
+      discard await node.publish(pubsubTopic, msg2)
+      discard await node.publish(pubsubTopic, msg3)
+      discard await node.publish(pubsubTopic, msg4)
+      
+      # Then the messages are received in both nodes
+      check:
+        await handlerFuture.withTimeout(3.seconds)
+        messageSeq == @[
+          (pubsubTopic, msg1),
+          (pubsubTopic, msg2),
+          (pubsubTopic, msg3),
+          (pubsubTopic, msg4)
+        ]
+        await otherHandlerFuture.withTimeout(3.seconds)
+        otherMessageSeq == @[
+          (pubsubTopic, msg1),
+          (pubsubTopic, msg2),
+          (pubsubTopic, msg3),
+          (pubsubTopic, msg4)
+        ]
+            
+      # Finally stop the other node
+      await allFutures(otherSwitch.stop(), otherNode.stop())
+
+  suite "Security and Privacy":
+    xasyncTest "Relay can receive messages after reboot":
+      # Given a second node connected to the first one
+      let
+        otherSwitch = newTestSwitch()
+        otherPeerManager = PeerManager.new(otherSwitch)
+        otherNode = await newTestWakuRelay(otherSwitch)
+
+      await allFutures(otherSwitch.start(), otherNode.start())
+      let 
+        otherRemotePeerInfo = otherSwitch.peerInfo.toRemotePeerInfo()
+        otherPeerId = otherRemotePeerInfo.peerId
+
+      check await peerManager.connectRelay(otherRemotePeerInfo)
+
+      # Given both are subscribed to the same pubsub topic
+      var otherHandlerFuture = newPushHandlerFuture()
+      proc otherSimpleFutureHandler(topic: PubsubTopic, message: WakuMessage) {.async, gcsafe.} =
+        otherHandlerFuture.complete((topic, message))
+
+      discard otherNode.subscribe(pubsubTopic, otherSimpleFutureHandler)
+      discard node.subscribe(pubsubTopic, simpleFutureHandler)
+      check:
+        node.subscribedTopics == pubsubTopicSeq
+        otherNode.subscribedTopics == pubsubTopicSeq
+      
+      await sleepAsync(500.millis)
+
+      # Given other node is stopped and restarted
+      await allFutures(otherSwitch.stop(), otherNode.stop())
+      await allFutures(otherSwitch.start(), otherNode.start())
+      # FIXME: Once stopped and started, nodes are not considered connected, nor do they reconnect after running connectRelay, as below
```

Comment:
Needs investigating.

Reply:
Does this actually depend on persistence of peer store? (Just a hunch, but maybe something to consider)...although peer store is probably note even enabled in this case

Reply:
I opened a ticket for this.